float32 x
float32 y
float32 length
float32 width
float32 magnitude
float32 azimuth
